client/session.go

```diff
@@ -3248,36 +3311,68 @@ func (it *peerBlocksIter) Next() bool {
 	return true
 }
 
+// Ensure streamBlocksResult implements blocksResult
+var _ blocksResult = (*bulkBlocksResult)(nil)
+
 type bulkBlocksResult struct {
 	sync.RWMutex
 	baseBlocksResult
-	result result.ShardResult
+	result         result.ShardResult
+	tagDecoderPool serialize.TagDecoderPool
+	idPool         ident.Pool
 }
 
 func newBulkBlocksResult(
 	opts Options,
 	resultOpts result.Options,
+	tagDecoderPool serialize.TagDecoderPool,
+	idPool ident.Pool,
 ) *bulkBlocksResult {
 	return &bulkBlocksResult{
 		baseBlocksResult: newBaseBlocksResult(opts, resultOpts),
 		result:           result.NewShardResult(4096, resultOpts),
+		tagDecoderPool:   tagDecoderPool,
+		idPool:           idPool,
 	}
 }
 
-func (r *bulkBlocksResult) addBlockFromPeer(id ident.ID, peer topology.Host, block *rpc.Block) error {
+func (r *bulkBlocksResult) addBlockFromPeer(
+	id ident.ID,
+	encodedTags checked.Bytes,
+	peer topology.Host,
+	block *rpc.Block,
+) error {
 	start := time.Unix(0, block.Start)
 	result, err := r.newDatabaseBlock(block)
 	if err != nil {
 		return err
 	}
 
+	var (
+		tags          ident.Tags
+		tryDecodeTags bool
```

Comment:
this seems more like its `triedToDecodeTags`

Reply:
Sure thing.